engine/src/main/scala/cromwell/webservice/PartialWorkflowSources.scala

```diff
@@ -239,11 +236,17 @@ object PartialWorkflowSources {
     JsObject(convertToMap reduce (_ ++ _))
   }
 
-  def convertStringToUrl(workflowUrl: String): ErrorOr[WorkflowUrl] = {
-    Try(new URL(workflowUrl)) match {
-      case Success(_) => workflowUrl.validNel
-      case Failure(e: IllegalUriException) => s"Invalid workflow url: ${e.getMessage}".invalidNel
-      case Failure(e) => s"Error while validating workflow url: ${e.getMessage}".invalidNel
+  def validateWorkflowUrl(workflowUrl: String): ErrorOr[WorkflowUrl] = {
+    def convertStringToUrl(workflowUrl: String): ErrorOr[WorkflowUrl] = {
+      Try(new URL(workflowUrl)) match {
+        case Success(_) => workflowUrl.validNel
+        case Failure(e) => s"Error while validating workflow url: ${e.getMessage}".invalidNel
+      }
+    }
+
+    workflowUrl.length match {
+      case l if l > MaxWorkflowUrlLength => s"Invalid workflow url: url has length $l, longer than the maximum allowed $MaxWorkflowUrlLength characters".invalidNel
```

Comment:
Minor point but this whole `match`/`case` could be restructured as a simpler `if (l > MaxWorkflowUrlLength) {} else {}`

Reply:
It sure can! 👍